server/service/src/sync/translations/program_requisition.rs

```diff
@@ -0,0 +1,242 @@
+
+        generate
+            .program_requisition_settings_rows
+            .into_iter()
+            .for_each(|program_requisition_setting| {
+                upserts.push(PullUpsertRecord::ProgramRequisitionSettings(
+                    program_requisition_setting,
+                ))
+            });
+
+        generate
+            .program_requisition_order_type_rows
+            .into_iter()
+            .for_each(|program_requisition_order_type| {
+                upserts.push(PullUpsertRecord::ProgramRequisitionOrderType(
+                    program_requisition_order_type,
+                ))
+            });
+
+        Ok(Some(IntegrationRecords::from_upserts(upserts)))
+    }
+
+    fn try_translate_pull_delete(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        let result = match_pull_table(sync_record).then(|| {
+            IntegrationRecords::from_delete(
+                &sync_record.record_id,
+                PullDeleteRecordTable::MasterList,
+            )
+        });
+
+        Ok(result)
+    }
+}
+
+#[derive(Clone)]
+struct GenerateRequisitionProgram {
+    pub program_row: ProgramRow,
+    pub program_requisition_settings_rows: Vec<ProgramRequisitionSettingsRow>,
+    pub program_requisition_order_type_rows: Vec<ProgramRequisitionOrderTypeRow>,
+}
+
+fn generate_requisition_program(
+    connection: &StorageConnection,
+    master_list: LegacyListMasterRow,
+) -> Result<Option<GenerateRequisitionProgram>, anyhow::Error> {
+    if master_list.is_program == false {
+        return Ok(None);
+    }
+
+    let program_settings = master_list.program_settings.clone().ok_or(anyhow::anyhow!(
+        "Program settings not found for program {}",
+        master_list.id
+    ))?;
+
+    let program_row = ProgramRow {
+        id: master_list.id.clone(),
+        master_list_id: master_list.id.clone(),
+        name: master_list.description.clone(),
+    };
+
+    let mut program_requisition_settings_rows = Vec::new();
+    let mut program_requisition_order_type_rows = Vec::new();
+
+    for tags in program_settings.store_tags.tags {
+        let name_tag = NameTagRowRepository::new(connection)
+            .find_one_by_name(&tags.0)?
+            .ok_or(anyhow::anyhow!(
+                "Name tag not found for program {}",
+                master_list.id
+            ))?;
+
+        let period_schedule = PeriodScheduleRowRepository::new(connection)
+            .find_one_by_name(&tags.1.period_schedule_name)?
+            .ok_or(anyhow::anyhow!(
+                "Period schedule not found for program {}",
+                master_list.id
+            ))?;
+
+        let program_requisition_settings_row = ProgramRequisitionSettingsRow {
+            // Concatenate the program id and name tag id to create a unique id
+            // instead of using uuid since easier to test this way.
```

Comment:
i think it would be easier to do the changes too (when central server config changes for a program)